### PR TITLE
Implementation of #107

### PR DESCRIPTION
**Pull request by Waldleufer**
#107

### CODE REVIEW

**Review by Waldleufer — COMMENTED** [review]
lgtm! 🙂

File: frontend/src/components/productGrid/ProductGridComponent.js [hunk 1]
```diff
@@ -30,13 +30,13 @@ function ProductGridComponent({ selectedCategory, selectedType }) {
         case 'generation':
             switch (selectedType) {
                 case 'products':
-                    products = preProducts.generation.products;
+                    products = preProducts?.generation?.products;
```
Comment (Waldleufer) [inline frontend/src/components/productGrid/ProductGridComponent.js:33]: genious!

**Review by Waldleufer — COMMENTED** [review]
finally!

File: frontend/src/interface/processBackendData.js [hunk 2]
```diff
@@ -72,3 +74,40 @@ export function processBackendData(data, callback) {
...
+ * This function segrigates the projects based on Generation, Transmission and Industrial APplications as mentioned in peojectDescription.
+ * @returns projects based on Generation, Transmission and Industrial APplications
+ */
+export function projectCategoryProcessing(projectDescription) {
+    if (
+        projectDescription === [] ||
+        projectDescription === undefined ||
+        projectDescription === null
+    ) {
+        return [categories.industrialApplications];
```
Comment (ScoutAtlas) [inline frontend/src/interface/processBackendData.js:90]: why if null do we return exactly industrial applications?

File: frontend/src/resources/globalConstants/SimaProProductCategories.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    generation: 'Generation',
+    transmission: 'Transmission',
+    industrialApplications: 'IndustrialApplication'
+};
+
+export const simaProTypes = {
+    product: 'Product',
+    solution: 'Solution',
+    service: 'Service'
+};
```
Comment (ScoutAtlas) [inline frontend/src/resources/globalConstants/SimaProProductCategories.js:14]: smart solution!

**Review by ScoutAtlas — COMMENTED** [review]
Very smart and elegant solution.

File: frontend/src/components/login/LoginComponent.js [hunk 4]
```diff
@@ -13,13 +15,22 @@ import React from 'react';
...
                     className='login'
                     action={slugs.categories}
-                    onSubmit={() => setState({ userIsLoggedIn: true })}
+                    onSubmit={() => {
+                        setState({ userIsLoggedIn: true });
+                        /*
+                         * useEffect declars the async function getProducts to be executed after the initial render and
+                         * hooks it so the Component reloads on change. At the moment the specified change is the selectedCategory.
+                         */
+                        dispatch(loadProjects());
```
Comment (MartinDuersch) [inline frontend/src/components/login/LoginComponent.js:32]: Good spot for the initial request and filling of state

File: frontend/src/components/productGrid/ProductGridComponent.js [hunk 5]
```diff
@@ -20,19 +20,63 @@ import { loadProjects } from 'store/actions/productAction';
...
+                    products = preProducts?.industrialApplication?.services;
+                    break;
+                default:
+                    break;
+            }
+            break;
+        default:
+            products = [];
+            break;
+    }
```
Comment (MartinDuersch) [inline frontend/src/components/productGrid/ProductGridComponent.js:79]: switch case is a bit short

File: frontend/src/interface/simaProInterface.js [hunk 6]
```diff
@@ -2,124 +2,188 @@
...
+                        productCategoriesAndTypes.industrialApplications.services.push(
+                            formattedProduct
+                        );
+                        break;
+                    default:
+                        break;
+                }
+                break;
+            default:
+                break;
```
Comment (MartinDuersch) [inline frontend/src/interface/simaProInterface.js:91]: barely long enough switch case :)

File: frontend/src/interface/processBackendData.js (hunk 2 above)
Comment (Waldleufer) [reply frontend/src/interface/processBackendData.js:90]: @SaiVarunVaranasi do you know? I am unsure. But if we change that, we need to handle `[]` cases I guess.

File: frontend/src/interface/simaProInterface.js [hunk 7]
```diff
@@ -2,124 +2,188 @@
...
+        } else {
+            let productSolutionService;
+
+            if (splittedString.includes(simaProTypes.product)) {
+                productSolutionService = types.product;
+            } else if (splittedString.includes(simaProTypes.solution)) {
+                productSolutionService = types.solution;
+            } else if (splittedString.includes(simaProTypes.service)) {
+                productSolutionService = types.service;
+            }
```
Comment (MartinDuersch) [inline frontend/src/interface/simaProInterface.js:133]: better than the switch case that was here before 🙃

File: frontend/src/resources/globalConstants/SimaProProductCategories.js (hunk 3 above)
Comment (MartinDuersch) [inline frontend/src/resources/globalConstants/SimaProProductCategories.js:14]: finally some well needed constants! 👍

File: frontend/src/assets/dummyData/dummyData.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+            return [
+                { modelID: 14, productID: productID, modelName: 'Allround Product 1' },
+                { modelID: 15, productID: productID, modelName: 'Allround Product 2' },
+                { modelID: 16, productID: productID, modelName: 'Allround Product 3' },
+                { modelID: 17, productID: productID, modelName: 'Allround Product 4' }
+            ];
+        default:
+            return [{ modelID: 42, productID: productID, modelName: productName }];
+    }
+}
```
Comment (MartinDuersch) [inline frontend/src/assets/dummyData/dummyData.js]: it has been 2,5 months and now the circle closes and were back at dummy data 😄

File: frontend/src/store/actions/productAction.js [hunk 9]
```diff
@@ -1,14 +1,20 @@
...
-      })
+        await getCategorizedProducts().then((products) => {
+            dispatch(processingGeneration(JSON.parse(JSON.stringify(products.generation))));
+            dispatch(processingTransmission(JSON.parse(JSON.stringify(products.transmission))));
+            dispatch(
+                processingIndustrialApplications(
+                    JSON.parse(JSON.stringify(products.industrialApplications))
+                )
+            );
+        });
```
Comment (MartinDuersch) [inline frontend/src/store/actions/productAction.js:16]: Nice reduxing, weird formatting

**Review by MartinDuersch — APPROVED** [review]
Greatly improved data management

File: frontend/src/store/actions/productAction.js (hunk 9 above)
Comment (Waldleufer) [reply frontend/src/store/actions/productAction.js:16]: 🤣

File: frontend/src/interface/processBackendData.js (hunk 2 above)
Comment (Waldleufer) [reply frontend/src/interface/processBackendData.js:90]: I changed that ;)

**Pull request merged**